js/src/hooks/usePolling.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import useApiFetchCallback from './useApiFetchCallback';
+import useCountdown from '.~/hooks/useCountdown';
+
+/**
+ * A hook for polling to an API Endpoint on intervals
+ *
+ * @param {import('@wordpress/api-fetch').APIFetchOptions} options Options to be forwarded to `apiFetch`.
+ * @param {number} [delay=20] Time in seconds to wait for each polling request
+ * @param {boolean} [stopOnResolved=false] If true. It stops the polling when the data is received
+ * @return {Object} Payload containing the result of the API call as well as a trigger function for starting the polling.
+ */
+export default function usePolling(
+	options,
+	delay = 10,
+	stopOnResolved = false
+) {
+	const { second, callCount, startCountdown } = useCountdown();
+	const [ fetch, { data } ] = useApiFetchCallback( options );
+
+	const start = useCallback(
+		( shouldRun = true ) => {
+			if ( ! shouldRun ) return;
+			const promise = fetch();
+			promise.finally( () => startCountdown( delay ) );
+			return promise;
+		},
+		[ fetch, startCountdown, delay ]
+	);
+
+	useEffect( () => {
+		if ( second === 0 && callCount > 0 ) {
+			start( ! data || ! stopOnResolved );
+		}
+	}, [ second, callCount, start, stopOnResolved ] );
```

Comment:
There is an eslint warning `react-hooks/exhaustive-deps`. Unfortunately, the [issue of duplicate API calls](https://github.com/woocommerce/google-listings-and-ads/pull/1733#discussion_r997974904) still appears if adding `data` to useEffect deps array.

![image](https://user-images.githubusercontent.com/17420811/196844134-2366afeb-bca3-48eb-9b9c-94c990e19c2f.png)

With the revision [9003b36](https://github.com/woocommerce/google-listings-and-ads/blob/9003b36ca713b699db55f8eca7948942aaa095ab/js/src/hooks/usePolling.js#L34-L38), replacing `stopOnResolved` and `data` in deps with a combined variable such as `const shouldRun = ! data || ! stopOnResolved;`, which is declared out of the useEffect scope, could be a solution.